### PR TITLE
Evaluate CIFAR-10 performance

### PR DESCRIPTION
**Pull request by quietlychris**
I was curious about evaluating the small ResNet's performance on CIFAR-10, so I gave it a shot using the `Test` set to evaluate the trained model. I can increase the accuracy from the baseline level on an untrained network up to around 34% accuracy, but it maxes out around there, which doesn't seem quite right;(I believe that you can actually hit the same on CIFAR-10 using a 4- or 5-layer fully-connected network. Any thoughts on how to improve accuracy, or if there's something I'm doing incorrectly in the implementation of the evaluation code that I've overlooked?

### CODE REVIEW

File: examples/train.rs [hunk 1]
```diff
@@ -44,14 +44,14 @@ fn main() {
...
-    for i_epoch in 0..1 {
+    for i_epoch in 0..10 {
         for (img, lbl) in train_data
             .shuffled(&mut rng)
             .map(preprocess)
             .batch(batch)
             .collate()
             .stack()
-            .take(30)
+            .take(128)
```
Comment (coreylowman) [inline examples/train.rs:54]: Have you tried using the whole dataset? I had the `.take()` in there to make profiling easier, but really we'd want to use the whole dataset for training

Comment (quietlychris) [reply examples/train.rs:54]: Ah, good point. Just ran it with both setting `.take(50_000)` and `// take()` (commenting it out entirely), both of which improve the results up to about 40%, but not more; training epochs are still set to 10.

Comment (coreylowman) [reply examples/train.rs:54]: Gotcha - the other thing is I kind of arbitrarily set up the small resnet architecture, I have no idea if it actually works haha. Might be worth trying to run it in pytorch to see if it achieves a different performance. That would give us a clue if there's a bug somewhere or just the network structure is bad
